src/main/scala/co/teapot/tempest/server/TempestDBServer.scala

```diff
@@ -44,19 +43,23 @@ class TempestDBServer(databaseClient: TempestDatabaseClient, config: TempestDBSe
   def nodes(graphName: String, sqlClause: String): util.List[lang.Long] =
     CollectionUtil.toJava(databaseClient.nodeIdsFiltered(graphName, sqlClause))
 
-  /** Returns the type of node reached after k steps along the given edge type. If the edge type
-    * is from nodeType1 to nodeType2, this will return nodeType1 if k is even, or nodeType2 if
-    * k is odd.
-    */
-  def kStepNodeType(edgeType: String, edgeDir: EdgeDir, k: Int): String = {
+
+  def loadEdgeConfig(edgeType: String): EdgeTypeConfig = {
     val edgeConfigFile = new File(config.graphConfigDirectoryFile, s"$edgeType.yaml")
     if (! edgeConfigFile.exists()) {
       throw new UndefinedGraphException(s"Graph config file ${edgeConfigFile.getCanonicalPath} not found")
     }
-    val edgeConfig = ConfigLoader.loadConfig[EdgeTypeConfig](edgeConfigFile)
+    ConfigLoader.loadConfig[EdgeTypeConfig](edgeConfigFile)
+  }
 
-    if ((edgeDir == EdgeDir.Out && k % 2 == 0) ||
-        (edgeDir == EdgeDir.In  && k % 2 == 1)) {
+  /** Returns the type of node reached after k steps along the given edge type. If the edge type
+    * is from nodeType1 to nodeType2, this will return nodeType1 if k is even, or nodeType2 if
+    * k is odd.
+    */
+  def kStepNodeType(edgeType: String, edgeDir: EdgeDir, k: Int): String = {
+    val edgeConfig = loadEdgeConfig(edgeType)
+    if ((edgeDir == EdgeDirOut && k % 2 == 0) ||
+        (edgeDir == EdgeDirIn  && k % 2 == 1)) {
       edgeConfig.getNodeType1
```

Comment:
How about we move away from NodeType1 and NodeType2 in the code? Say SourceNodeType and TargetNodeType?